### PR TITLE
Editor features

### LINKED ISSUE #28: [implementer-tools] Make arrays prettier
Right now, arrays are presented as long strings when viewing in the config. Editing of arrays is also challenging because it simply creates a textarea for editing the string. This is prone to mistakes and hard to read. Could we come up with a prettier way for both editing and viewing this data within the configuration?

### PR DESCRIPTION
**Pull request by brandones**
This adds support for a variety of different types of values in the editor, including arrays. It's a major overhaul of the editor. Apologies for the big PR.

demo

A few important things that remain after this PR:
- support for the "order" and "configure" fields of extensions
- getting the array tests to work
- writing array-of-object tests
- support for freeform objects
- better design

Also, right now values don't quite update properly when they are changed, but that's an issue apart from the editor. That's just about esm-config needing to be refactored to use unistore.

Closes #28

### CODE REVIEW

**Comment by denniskigen** [comment]
Great work @brandones

**Comment by brandones** [comment]
Thanks @denniskigen ! Do you want to review/approve? Sorry I know it's a big one.

**Comment by denniskigen** [comment]
Yeah, reviewing it

File: packages/esm-implementer-tools-app/src/configuration/configuration.test.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
+import React from "react";
```
Comment (denniskigen) [inline packages/esm-implementer-tools-app/src/configuration/configuration.test.tsx:1]: Great tests!

**Review by denniskigen — APPROVED** [review]
Phenomenal work Brandon. The implementer tools panel is really taking shape.

File: packages/esm-implementer-tools-app/src/configuration/configuration.component.tsx [hunk 2]
```diff
@@ -43,59 +41,55 @@ export default function Configuration(props: ConfigurationProps) {
...
+          <a
+            className={styles.downloadLink}
+            download="temporary_config.json"
+            href={window.URL.createObjectURL(tempConfigObjUrl)}
+          >
+            Download Temporary Config
+          </a>
+        </Button>
+      </div>
+      <div className={styles.mainContent}>
```
[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-implementer-tools-app/src/configuration/configuration.component.tsx:84]: It went through a few iterations—getting rid of the grid made more sense when I had the descriptions in a floating box on the right. Now it's just an even split on the screen between the config tree and descriptions, and Grid would be fine. But it really wouldn't contribute anything here, either :shrug:

**Comment by brandones** [comment]
Thanks a bunch @denniskigen !

**Pull request merged**